overlord/snapstate/backend/link.go

```diff
@@ -191,6 +191,18 @@ func (b Backend) generateWrappers(s *snap.Info, linkCtx LinkContext) error {
 	}
 	cleanupFuncs = append(cleanupFuncs, wrappers.RemoveSnapBinaries)
 
+	// check DaemonStartup settings and do not start services on first
+	// install that are inhibited
+	if linkCtx.FirstInstall {
```

Comment:
I think we need to decide on this, my thinking was that the decision shouldn't be based whether we are installing the snap itself, but whether the service exists atm or not yet, that's the "installl" of the service, not the snap. We might need to think whether that impacts refresh-mode behavior or not.

Reply:
Agreed, I think this code is wrong, I added a XXX here and in the spread test how this should work so that upgrades also work well.

Reply:
The code now looks at the previous installed snap and only applies "install-mode: disable" to new services.